llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -756,9 +756,10 @@ class OpLowerer {
       case Intrinsic::not_intrinsic:
         continue;
       default: {
-        DiagnosticInfoUnsupported Diag(
-            F, "Unsupported intrinsic for DXIL lowering");
-        M.getContext().diagnose(Diag);
+        std::string Msg =
+            formatv("Unsupported intrinsic {0} for DXIL lowering", F.getName())
+                .str();
```

Comment:
Might as well use SmallString here:
```suggestion
        SmallString<128> Msg =
            formatv("Unsupported intrinsic {0} for DXIL lowering", F.getName());
```